llvm/include/llvm/Transforms/Utils/LoopUtils.h

```diff
@@ -330,6 +330,10 @@ getLoopEstimatedTripCount(Loop *L,
 bool setLoopEstimatedTripCount(Loop *L, unsigned EstimatedTripCount,
                                unsigned EstimatedLoopInvocationWeight);
 
+// Return true if loop's estimated trip count based on branch weight metadata is
+// infinite, i.e. weight ratio is X:0.
+bool isInfiniteTripCount(Loop *L);
```

Comment:
`const Loop& L`. `const` because you shouldn't need to mutate `L`, pass-by-reference to signal `nullptr` isn't something we expect handling.